### PR TITLE
feat: automate PR for dumping kubelet flags

### PR DESCRIPTION
**Pull request by anujmaheshwari1**
**What type of PR is this?**

**What this PR does / why we need it**:
This PR is the next stage is generating kubelet flags. It helps create a PR with the changed kubelet-flags.json file, which will later be used for stripping removed flags.

Sample PR: https://github.com/Azure/AgentBaker/commit/a877c78f7328e04b6a00cfdd90e81662cf4d4937
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: .github/workflows/generate-kubelet-flags.yaml [hunk 1]
```diff
@@ -9,6 +9,27 @@ jobs:
...
+        git commit -m"feat: generate kubelet flags for supported versions in manifest.json"
+        git push -u origin kubeletFlags/$timestamp -f
+        curl \
+          -X POST \
+          https://api.github.com/repos/Azure/AgentBaker/pulls \
+          -d '{"head" : "'kubeletFlags/$timestamp'", "base" : "master", "title" : "bump: automated PR for updating kubelet flags files"}' \
+          -u "alexeldeib:$github_token"
+      env:
+        github_token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+        timestamp: ${{ env.TIMESTAMP }}
```
Comment (alexeldeib) [inline .github/workflows/generate-kubelet-flags.yaml:35]: is this not already in env?

Comment (anujmaheshwari1) [reply .github/workflows/generate-kubelet-flags.yaml:35]: no, I have to pipe it down from the previous stage where I parsed it.

Comment (alexeldeib) [reply .github/workflows/generate-kubelet-flags.yaml:35]: but you don't need to reset timestamp from env.TIMESTAMP, right? 

https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

**Review by alexeldeib — APPROVED** [review]

File: .github/workflows/generate-kubelet-flags.yaml (hunk 1 above)
Comment (anujmaheshwari1) [reply .github/workflows/generate-kubelet-flags.yaml]: resolved

File: .github/workflows/generate-kubelet-flags.yaml [hunk 2]
```diff
@@ -9,6 +9,26 @@ jobs:
...
+        echo "TIMESTAMP=$TIMESTAMP" >> $GITHUB_ENV
+        git checkout -b kubeletFlags/$TIMESTAMP
+    - name: Generate valid kubelet flags per supported k8s versions
+      run: |
         make generate-kubelet-flags
-      name: Generate valid kubelet flags per supported k8s versions
\ No newline at end of file
+    - name: Create PR for modified kubelet flag files
+      run: |
+        git config --global user.email "aleldeib@microsoft.com"
```
[the remark this answers is not in the document]
Comment (anujmaheshwari1) [reply .github/workflows/generate-kubelet-flags.yaml:22]: the PAT already configured in the repo is Ace's for fuzzing. asked his permission before using it.

**Review by juan-lee — APPROVED** [review]

**Pull request merged**